payments-core/src/main/java/com/stripe/android/model/ConfirmPaymentIntentParams.kt

```diff
@@ -320,6 +320,28 @@ data class ConfirmPaymentIntentParams internal constructor(
         internal const val PARAM_SOURCE_DATA: String = "source_data"
         private const val PARAM_SOURCE_ID = "source"
 
+        /**
+         * Create the parameters necessary for confirming a [PaymentIntent] based on its [clientSecret]
+         * and [paymentMethodType]
+         *
+         * Use this initializer for PaymentIntents that already have a PaymentMethod attached.
+         *
+         * @param clientSecret client secret from the PaymentIntent that is to be confirmed
+         * @param paymentMethodType the known type of the PaymentIntent's attached PaymentMethod
+         */
+        @JvmStatic
+        fun create(
+            clientSecret: String,
+            paymentMethodType: PaymentMethod.Type
+        ): ConfirmPaymentIntentParams {
+            return ConfirmPaymentIntentParams(
+                clientSecret = clientSecret,
+                // infers default [MandateDataParams] based on the attached [paymentMethodType]
+                mandateData = MandateDataParams(MandateDataParams.Type.Online.DEFAULT)
+                    .takeIf { paymentMethodType.requiresMandate }
```

Comment:
Although this is tested, we could extract this logic out if we don't feel like the `PaymentMethod -> Default Mandate Params` mapping belongs here. Open to suggestions here.

Reply:
Won't the code in [line 196](https://github.com/stripe/stripe-android/blob/master/payments-core/src/main/java/com/stripe/android/model/ConfirmPaymentIntentParams.kt#L196) work for this already?
```
if (paymentMethodCreateParams?.type?.requiresMandate == true && mandateId == null) {
    // Populate with default "online" MandateData
    MandateDataParams(MandateDataParams.Type.Online.DEFAULT).toParamMap()
}
```

Reply:
For the above to work, we'd need to pass a `paymentMethodCreateParams` object, which is used when creating the payment method at confirm time.

This flow is for confirmation of already created payment methods, but for the above to work we'd need to construct the entire payment method object and use [createWithPaymentMethodCreateParams](https://github.com/stripe/stripe-android/blob/master/payments-core/src/main/java/com/stripe/android/model/ConfirmPaymentIntentParams.kt#L403).

Reply:
Another alternative we thought about was introducing a transient `PaymentMethod.Type` field into `ConfirmParams` and modifying the [logic to build the payment method params](https://github.com/stripe/stripe-android/blob/master/payments-core/src/main/java/com/stripe/android/model/ConfirmPaymentIntentParams.kt#L173), but doesn't sound like a very reasonable approach as the payment method type is not part of the params to be sent, but instead used to infer default mandate data.

Reply:
That makes sense, I worry about this logic being replicated in multiple places, but this seems like a special case.